File: NVR_Client_App/90116/ActiveX-Sample/Sample Program/PS-API/Visual Basic 2012/013_Overlay/Form1.vb
Public Class Form1
    '****************************************************************
    '* Define
    '****************************************************************
    'LiveStatus 0:Start 1:Live
    Public gPlayStatus As Long = 0


    '****************************************************************
    '* Function Name        : Load
    '* Overview             : Load PSAPI & Initialize
    '****************************************************************
    Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        'Show the log window
        With Form2
            .Show()
        End With

        '----------------------------------------
        'Set properties
        '----------------------------------------
        With AxipropsapiCtrl1
            'Set properties to connect the device
            .IPAddr = "192.168.0.10"
            .DeviceType = 2
            .HttpPort = 80
            .UserName = "admin"
            .Password = "12345"

            'Set properties for image format
            .StreamFormat = 0
            .JPEGResolution = 640
            .MPEG4Resolution = 640
            .H264Resolution = 640

            'Set properties for event
            .OnErrorEnable = 1
            .OnDevStatusEnable = 0
            .OnRecStatusEnable = 0
            .OnPlayStatusEnable = 1
            .OnImageRefreshEnable = 0
            .OnRecordStatusEnable = 0
            .OnOpStatusEnable = 1
            .OnAlmStatusEnable = 0

            .OnRecStatusCBEnable = 0
            .OnSearchCBEnable = 0
            .OnSearchExCBEnable = 0
            .OnPlayStatusCBEnable = 0
            .OnOpStatusCBEnable = 0
            .OnAlmStatusCBEnable = 0
            .OnFtpStatusCBEnable = 0
        End With

        '----------------------------------------
        'Connect to the device
        '----------------------------------------
        Dim lRet As Long
        Dim lBlockmode As Long
        Dim lChannel As Long

        lRet = AxipropsapiCtrl1.Open
        Logging("[Function] Open:" & CStr(lRet))

        If lRet > -1 Then
	        lChannel = 1
	        lBlockmode = 1
            lRet = AxipropsapiCtrl1.PlayLive(lChannel, lBlockmode)
            Logging("[Function] PlayLive(Start):" & CStr(lRet))

            If lRet = 0 Then
                gPlayStatus = 1
            Else
                AxipropsapiCtrl1.Close()
            End If
        End If

        ' Added Ver5.0.1.0
        TextBox1.Text = "C:\temp\test.bmp"
        TextBox1.AllowDrop = True

    End Sub

    '****************************************************************
    '* Function Name        : FormClosed
    '* Overview             : Destroy log window
    '****************************************************************
    Private Sub Form1_FormClosed(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles Me.FormClosed
        '----------------------------------------
        'Disconnect to the device
        '----------------------------------------
        Dim lRet As Long
        Dim lCommand As Long
        Dim lSpeed As Long
        Dim lBlockmode As Long

        'Check the play status
        If gPlayStatus = 1 Then
            lCommand = 1
            lSpeed = 1
            lBlockmode = 0
            lRet = AxipropsapiCtrl1.PlayControl(lCommand, lSpeed, lBlockmode)
            Logging("[Function] PlayLive(Stop):" & CStr(lRet))

            'Close connection to the device
            AxipropsapiCtrl1.Close()
            Logging("[Function] Close")

            'ClearImage
            AxipropsapiCtrl1.ClearImage()
        End If

        '----------------------------------------
        'Set properties
        '----------------------------------------
        With AxipropsapiCtrl1
            'Set properties for event
            .OnErrorEnable = 0
            .OnDevStatusEnable = 0
            .OnRecStatusEnable = 0
            .OnPlayStatusEnable = 0
            .OnImageRefreshEnable = 0
            .OnRecordStatusEnable = 0
            .OnOpStatusEnable = 0
            .OnAlmStatusEnable = 0

            .OnRecStatusCBEnable = 0
            .OnSearchCBEnable = 0
            .OnSearchExCBEnable = 0
            .OnPlayStatusCBEnable = 0
            .OnOpStatusCBEnable = 0
            .OnAlmStatusCBEnable = 0
            .OnFtpStatusCBEnable = 0
        End With

        With Form2
            .Close()
        End With
    End Sub

    '****************************************************************
    '* Function Name        : Logging
    '* Overview             : Output Logs
    '****************************************************************
    Private Sub Logging(ByVal str As String)
        With Form2
            If .txt_Log.Text <> "" Then
                .txt_Log.Text = .txt_Log.Text & vbCrLf
            End If
            .txt_Log.Text = .txt_Log.Text & str
        End With
    End Sub

    '****************************************************************
    '* Function Name        : ShowResult
    '* Overview             : Output list Search result
    ' ****************************************************************
    Private Sub ShowResult()
        'Set Search Result to list
    End Sub

    '****************************************************************
    '* Function Name        : ShowResultEx
    '* Overview             : Output list SearchEx/VMDSearchEx result
    ' ****************************************************************
    Private Sub ShowResultEx()
        'Set SearchEx Result to list
    End Sub

    '****************************************************************
    '* Function Name        : chk_drawTitle_CheckedChanged      Modefied Ver5.0.1.0
    '* Overview             : Draw Title
    '****************************************************************
    Private Sub chk_drawTitle_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles chk_drawTitle.CheckedChanged
        'Define variables
        Dim lRet As Long
        Dim lId As Long
        Dim lCommand As Long
        Dim strText As String
        Dim lxPosition As Long
        Dim lyPosition As Long
        Dim lAlign As Long
        Dim strFont As String
        Dim lFontSize As Long
        Dim lForeColor As Long
        Dim lBorderColor As Long
        Dim lStyle As Long
        Dim Transmittance As Integer

        strText = "PS-API ActiveX"
        lxPosition = 100
        lyPosition = 100
        lAlign = 0
        strFont = "MS UI Gothic"
        lFontSize = 24
        lForeColor = 0
        lBorderColor = 16777215
        lStyle = 2
        Transmittance = &H7F

        If (chk_drawTitle.Checked = True) Then
            'Display Title
            lId = 1
            lCommand = 1
            'Change method(TitleOperation->TitleOperationEx)
            lRet = AxipropsapiCtrl1.TitleOperationEx(lId, lCommand, strText, lxPosition, lyPosition, lAlign, strFont, lFontSize, lForeColor, lBorderColor, lStyle, Transmittance)
            Logging("[Function] TitleOperationEx (ON):" & CStr(lRet))
        Else
            'Not display Title
            lId = 1
            lCommand = 0
            'Change method(TitleOperation->TitleOperationEx)
            lRet = AxipropsapiCtrl1.TitleOperationEx(lId, lCommand, strText, lxPosition, lyPosition, lAlign, strFont, lFontSize, lForeColor, lBorderColor, lStyle, Transmittance)
            Logging("[Function] TitleOperationEx (OFF):" & CStr(lRet))
        End If
    End Sub

    '****************************************************************
    '* Function Name        : chk_drawBox_CheckedChanged        Modefied Ver5.0.1.0
    '* Overview             : Draw Box
    '****************************************************************
    Private Sub chk_drawBox_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles chk_drawBox.CheckedChanged
        'Define variables
        Dim lRet As Long
        Dim lId As Long
        Dim lCommand As Long
        Dim lColor As Long
        Dim lSize As Long
        Dim lxTopLeft As Long
        Dim lyTopLeft As Long
        Dim lxBottomRight As Long
        Dim lyBottomRight As Long
        Dim Transmittance As Integer

        lColor = 255
        lSize = 3
        lxTopLeft = 200
        lyTopLeft = 200
        lxBottomRight = 300
        lyBottomRight = 300
        Transmittance = &H7F

        If (chk_drawBox.Checked = True) Then
            'Display Box
            lId = 1
            lCommand = 2
            'Change method(BoxOperation->BoxOperationEx)
            lRet = AxipropsapiCtrl1.BoxOperationEx(lId, lCommand, lColor, lSize, lxTopLeft, lyTopLeft, lxBottomRight, lyBottomRight, Transmittance)
            Logging("[Function] BoxOperationEx (ON):" & CStr(lRet))
        Else
            'Not display Box
            lId = 1
            lCommand = 0
            'Change method(BoxOperation->BoxOperationEx)
            lRet = AxipropsapiCtrl1.BoxOperationEx(lId, lCommand, lColor, lSize, lxTopLeft, lyTopLeft, lxBottomRight, lyBottomRight, Transmittance)
            Logging("[Function] BoxOperationEx (OFF):" & CStr(lRet))
        End If
    End Sub

    '****************************************************************
    '* Function Name        : chk_BMPDraw_CheckedChanged        Added Ver5.0.1.0
    '* Overview             : Transparent bitmap image processing
    '****************************************************************
    Private Sub chk_BMPDraw_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles chk_BMPDraw.CheckedChanged
        Dim Ret As Integer
        Dim Id As Integer
        Dim Command As Integer
        Dim FileName As String
        Dim XPos As Integer
        Dim YPos As Integer
        Dim MaskColor As Integer
        Dim Transmittance As Integer

        Id = 1                  'ID for management
        XPos = 320              'X position of displayed transparent
        YPos = 240              'Y position of displayed transparent
        If chk_BMPDraw.Checked Then
            'Transparent bitmap
            FileName = TextBox1.Text
            Command = 1             'Bitmap display
            MaskColor = &HFF00FF    'Bitmap's masked color
            Transmittance = &H7F    'Transmittance
            Ret = AxipropsapiCtrl1.BitmapOperationEx(Id, Command, FileName, XPos, YPos, MaskColor, Transmittance)
            Logging("[Function] BitmapOperationEx (ON," & FileName & "):" & Ret.ToString())
        Else
            'Not transparent bitmap
            Command = 0             'Bitmap non display
            MaskColor = &H0         'Bitmap's masked color
            Transmittance = &H7F    'Transmittance
            Ret = AxipropsapiCtrl1.BitmapOperationEx(Id, Command, String.Empty, 0, 0, MaskColor, Transmittance)
            Logging("[Function] BitmapOperationEx (OFF):" & Ret.ToString())
        End If
    End Sub

    '****************************************************************
    '* Function Name        : DragEnter                         Added Ver5.0.1.0
    '* Overview             : Drag the files and only valid
    '****************************************************************
    Private Sub TextBox1_DragEnter(ByVal sender As System.Object, ByVal e As System.Windows.Forms.DragEventArgs) Handles TextBox1.DragEnter
        If e.Data.GetDataPresent(DataFormats.FileDrop) Then
            e.Effect = DragDropEffects.Copy
        Else
            e.Effect = DragDropEffects.None
        End If
    End Sub

    '****************************************************************
    '* Function Name        : DragDrop                          Added Ver5.0.1.0
    '* Overview             : Drop a single file to view
    '****************************************************************
    Private Sub TextBox1_DragDrop(ByVal sender As System.Object, ByVal e As System.Windows.Forms.DragEventArgs) Handles TextBox1.DragDrop
        TextBox1.Text = e.Data.GetData(DataFormats.FileDrop)(0)
    End Sub

    '****************************************************************
    '* Function Name        : OnError
    '* Overview             : OnError listener event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnError(ByVal sender As System.Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnErrorEvent) Handles AxipropsapiCtrl1.OnError
        'Output Logs
        Logging("[OnError] errorCode[" & CStr(e.errorCode) & "] description[" & e.description & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnDevStatus
    '* Overview             : OnDevStatus listener event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnDevStatus(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnDevStatusEvent) Handles AxipropsapiCtrl1.OnDevStatus
        'Output Logs
        Logging("[OnDevStatus] channel[" & CStr(e.channel) & "] status[" & CStr(e.status) & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnRecStatus
    '* Overview             : OnRecStatus listener event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnRecStatus(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnRecStatusEvent) Handles AxipropsapiCtrl1.OnRecStatus
        'Output Logs
        Logging("[OnRecStatus] channel[" & CStr(e.channel) & "] status[" & CStr(e.status) & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnPlayStatus
    '* Overview             : OnPlayStatus listener event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnPlayStatus(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnPlayStatusEvent) Handles AxipropsapiCtrl1.OnPlayStatus
        'Output Logs
        Logging("[OnPlayStatus] channel[" & CStr(e.channel) & "] status[" & CStr(e.status) & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnRecordStatus
    '* Overview             : OnRecordStatus listener event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnRecordStatus(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnRecordStatusEvent) Handles AxipropsapiCtrl1.OnRecordStatus
        'Output Logs
        Logging("[OnRecordStatus] type[" & CStr(e.recType) & "] timeDate[" & e.timeDate & "] nextRecTime[" & e.nextRecTime & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnImageRefresh
    '* Overview             : OnImageRefresh listener event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnImageRefresh(ByVal sender As Object, ByVal e As System.EventArgs) Handles AxipropsapiCtrl1.OnImageRefresh
        'Output Logs
        Logging("[OnImageRefresh] No argument.")
    End Sub

    '****************************************************************
    '* Function Name        : OnOpStatus
    '* Overview             : OnOpStatus listener event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnOpStatus(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnOpStatusEvent) Handles AxipropsapiCtrl1.OnOpStatus
        'Output Logs
        Logging("[OnOpStatus] channel[" & CStr(e.channel) & "] status[" & CStr(e.status) & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnAlmStatus
    '* Overview             : OnAlmStatus listener event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnAlmStatus(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnAlmStatusEvent) Handles AxipropsapiCtrl1.OnAlmStatus
        'Output Logs
        Logging("[OnAlmStatus] channel[" & CStr(e.channel) & "] type[" & CStr(e.alarmtype) & "] timeDate[" & e.timeDate & "] status[" & CStr(e.status) & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnRecStatusCB
    '* Overview             : OnRecStatusCB callback event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnRecStatusCB(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnRecStatusCBEvent) Handles AxipropsapiCtrl1.OnRecStatusCB
        'Output Logs
        Logging("[OnRecStatusCB] status[" & CStr(e.status) & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnSearchCB
    '* Overview             : OnSearchCB callback event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnSearchCB(ByVal sender As Object, ByVal e As System.EventArgs) Handles AxipropsapiCtrl1.OnSearchCB
        'Output Logs
        Logging("[OnSearchCB] Show Search result.")
        ShowResult()
    End Sub

    '****************************************************************
    '* Function Name        : OnSearchExCB
    '* Overview             : OnSearchExCB callback event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnSearchExCB(ByVal sender As Object, ByVal e As System.EventArgs) Handles AxipropsapiCtrl1.OnSearchExCB
        'Output Logs
        Logging("[OnSearchExCB] Show Search result.")
        ShowResultEx()
    End Sub

    '****************************************************************
    '* Function Name        : OnPlayStatusCB
    '* Overview             : OnPlayStatusCB callback event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnPlayStatusCB(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnPlayStatusCBEvent) Handles AxipropsapiCtrl1.OnPlayStatusCB
        'Output Logs
        Logging("[OnPlayStatusCB] status[" & CStr(e.status) & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnOpStatusCB
    '* Overview             : OnOpStatusCB callback event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnOpStatusCB(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnOpStatusCBEvent) Handles AxipropsapiCtrl1.OnOpStatusCB
        'Output Logs
        Logging("[OnOpStatusCB] status[" & CStr(e.status) & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnAlmStatusCB
    '* Overview             : OnAlmStatusCB callback event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnAlmStatusCB(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnAlmStatusCBEvent) Handles AxipropsapiCtrl1.OnAlmStatusCB
        'Output Logs
        Logging("[OnAlmStatusCB] status[" & CStr(e.status) & "]")
    End Sub

    '****************************************************************
    '* Function Name        : OnFtpStatusCB
    '* Overview             : OnFtpStatusCB callback event
    '****************************************************************
    Private Sub AxipropsapiCtrl1_OnFtpStatusCB(ByVal sender As Object, ByVal e As AxIPROPSAPILib._IipropsapiCtrlEvents_OnFtpStatusCBEvent) Handles AxipropsapiCtrl1.OnFtpStatusCB
        'Output Logs
        Logging("[OnFtpStatusCB] status[" & CStr(e.status) & "]")
    End Sub

End Class

File: NVR_Client_App/90116/ActiveX-Sample/Sample Program/PS-API/Visual CSharp/013_Overlay/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace SimpleSample_Overlay
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        //---------------------------------------------------------------------
        // Define variables
        //---------------------------------------------------------------------
        private Form2 frmForm2;

        int m_PlayStatus = 0;    //LiveStatus 0:Stop 1:Live

        //---------------------------------------------------------------------
        // Function Name        : Load
        // Overview             : Load PSAPI and Initialize
        //---------------------------------------------------------------------
        private void Form1_Load(object sender, EventArgs e)
        {
            //Show the log window
            frmForm2 = new Form2();
            frmForm2.Show();

            //----------------------------------------
            //Set properties
            //----------------------------------------
            //Set properties to connect the device
            axipropsapiCtrl1.IPAddr = "192.168.0.10";
            axipropsapiCtrl1.DeviceType = 2;
            axipropsapiCtrl1.HttpPort = 80;
            axipropsapiCtrl1.UserName = "admin";
            axipropsapiCtrl1.Password = "12345";

            //Set properties for display area
            axipropsapiCtrl1.StreamFormat = 0;
            axipropsapiCtrl1.JPEGResolution = 640;
            axipropsapiCtrl1.MPEG4Resolution = 640;
            axipropsapiCtrl1.H264Resolution = 640;

            //Set properties for event
            axipropsapiCtrl1.OnErrorEnable = 1;
            axipropsapiCtrl1.OnDevStatusEnable = 0;
            axipropsapiCtrl1.OnRecStatusEnable = 0;
            axipropsapiCtrl1.OnPlayStatusEnable = 1;
            axipropsapiCtrl1.OnImageRefreshEnable = 0;
            axipropsapiCtrl1.OnRecordStatusEnable = 0;
            axipropsapiCtrl1.OnOpStatusEnable = 1;
            axipropsapiCtrl1.OnAlmStatusEnable = 0;

            axipropsapiCtrl1.OnRecStatusCBEnable = 0;
            axipropsapiCtrl1.OnSearchCBEnable = 0;
            axipropsapiCtrl1.OnSearchExCBEnable = 0;
            axipropsapiCtrl1.OnPlayStatusCBEnable = 0;
            axipropsapiCtrl1.OnOpStatusCBEnable = 0;
            axipropsapiCtrl1.OnAlmStatusCBEnable = 0;
            axipropsapiCtrl1.OnFtpStatusCBEnable = 0;

            //----------------------------------------
            //Connect to the device
            //----------------------------------------
            int iRet;
            int iChannel;
            int iBlockMode;

            iRet = axipropsapiCtrl1.Open();
            Logging("[Function] Open:" + iRet.ToString());

            if (iRet > -1)
            {
                iChannel = 1;
                iBlockMode = 1;
                iRet = axipropsapiCtrl1.PlayLive(iChannel, iBlockMode);
                Logging("[Function] PlayLive(Start):" + (iRet));

                if (iRet == 0)
                {
                    m_PlayStatus = 1;
                }
                else
                {
                    axipropsapiCtrl1.Close();
                    Logging("[Function] Close");
                }
            }
            // Added Ver5.0.1.0
            textBox1.Text = "C:\\temp\\test.bmp";
            textBox1.AllowDrop = true;
        }

        //---------------------------------------------------------------------
        // Function Name        : FormClosed
        // Overview             : Destroy log window
        //---------------------------------------------------------------------
        private void Form1_FormClosed(object sender, FormClosedEventArgs e)
        {

            int iRet;
            int iCommand;
            int iSpeed;
            int iBlockMode;

            if(m_PlayStatus == 1)
            {
                iCommand = 1;
                iSpeed = 1;
                iBlockMode = 0;
                iRet = axipropsapiCtrl1.PlayControl(iCommand, iSpeed, iBlockMode);
                Logging("[Function] PlayControl(Stop):" + (iRet));

                axipropsapiCtrl1.Close();
                Logging("[Function] Close");

                axipropsapiCtrl1.ClearImage();
                Logging("[Function] ClearImage");
            }

            //----------------------------------------
            //Set properties
            //----------------------------------------
            //Set properties for event
            axipropsapiCtrl1.OnErrorEnable = 0;
            axipropsapiCtrl1.OnDevStatusEnable = 0;
            axipropsapiCtrl1.OnRecStatusEnable = 0;
            axipropsapiCtrl1.OnPlayStatusEnable = 0;
            axipropsapiCtrl1.OnImageRefreshEnable = 0;
            axipropsapiCtrl1.OnRecordStatusEnable = 0;
            axipropsapiCtrl1.OnOpStatusEnable = 0;
            axipropsapiCtrl1.OnAlmStatusEnable = 0;

            axipropsapiCtrl1.OnRecStatusCBEnable = 0;
            axipropsapiCtrl1.OnSearchCBEnable = 0;
            axipropsapiCtrl1.OnSearchExCBEnable = 0;
            axipropsapiCtrl1.OnPlayStatusCBEnable = 0;
            axipropsapiCtrl1.OnOpStatusCBEnable = 0;
            axipropsapiCtrl1.OnAlmStatusCBEnable = 0;
            axipropsapiCtrl1.OnFtpStatusCBEnable = 0;

            frmForm2.Close();

        }

        //---------------------------------------------------------------------
        // Function Name        : Logging
        // Overview             : Output Logs
        //---------------------------------------------------------------------
        private void Logging(String str)
        {
            if (frmForm2.textLog.Text != "")
            {
                frmForm2.textLog.Text = frmForm2.textLog.Text + "\r\n";
            }
            frmForm2.textLog.Text = frmForm2.textLog.Text + str;
        }

        //---------------------------------------------------------------------
        // Function Name        : ShowResult
        // Overview             : Output list Search result
        //---------------------------------------------------------------------
        private void ShowResult()
        {
            //Set Search Result to list
        }

        //---------------------------------------------------------------------
        // Function Name        : ShowResultEx
        // Overview             : Output list SearchEx/VMDSearchEx result
        //---------------------------------------------------------------------
        private void ShowResultEx()
        {
            //Set SearchEx Result to list
        }

        //---------------------------------------------------------------------
        // Function Name        : chk_drawTitle_CheckedChanged      Modefied Ver5.0.1.0
        // Overview             : Draw Title
        //---------------------------------------------------------------------
        private void chk_drawTitle_CheckedChanged(object sender, EventArgs e)
        {
            //Define variables
            int iRet;
            int iId;
            int iCommand;
            string strText;
            int ixPosition;
            int iyPosition;
            int iAlign;
            string strFont;
            int iFontSize;
            int iForeColor;
            int iBorderColor;
            int iStyle;
            int iTransmittance;

            strText = "PS-API ActiveX";
            ixPosition = 100;
            iyPosition = 100;
            iAlign = 0;
            strFont = "MS UI Gothic";
            iFontSize = 24;
            iForeColor = 0;
            iBorderColor = 16777215;
            iStyle = 2;
            iTransmittance = 0x7F;

            if (chk_drawTitle.Checked == true)
            {
                //Display Title
                iId = 1;
                iCommand = 1;
                iRet = axipropsapiCtrl1.TitleOperationEx(iId, iCommand, strText, ixPosition, iyPosition, iAlign, strFont, iFontSize, iForeColor, iBorderColor, iStyle, iTransmittance);
                Logging("[Function] TitleOperation (ON):" + iRet.ToString());
            }
            else
            {
                //Not display Title
                iId = 1;
                iCommand = 0;
                iRet = axipropsapiCtrl1.TitleOperationEx(iId, iCommand, strText, ixPosition, iyPosition, iAlign, strFont, iFontSize, iForeColor, iBorderColor, iStyle, iTransmittance);
                Logging("[Function] TitleOperation (OFF):" + iRet.ToString());
            }
        }

        //---------------------------------------------------------------------
        // Function Name        : chk_drawBox_CheckedChanged        Modefied Ver5.0.1.0
        // Overview             : Draw Box
        //---------------------------------------------------------------------
        private void chk_drawBox_CheckedChanged(object sender, EventArgs e)
        {
            //Define variables
            int iRet;
            int iId;
            int iCommand;
            int iColor;
            int iSize;
            int ixTopLeft;
            int iyTopLeft;
            int ixBottomRight;
            int iyBottomRight;
            int iTransmittance;

            iColor = 255;
            iSize = 3;
            ixTopLeft = 200;
            iyTopLeft = 200;
            ixBottomRight = 300;
            iyBottomRight = 300;
            iTransmittance = 0x7F;

            if (chk_drawBox.Checked == true)
            {
                //Display Box
                iId = 1;
                iCommand = 2;
                iRet = axipropsapiCtrl1.BoxOperationEx(iId, iCommand, iColor, iSize, ixTopLeft, iyTopLeft, ixBottomRight, iyBottomRight, iTransmittance);
                Logging("[Function] BoxOperation (ON):" + iRet.ToString());
            }
            else
            {
                //Not display Box
                iId = 1;
                iCommand = 0;
                iRet = axipropsapiCtrl1.BoxOperationEx(iId, iCommand, iColor, iSize, ixTopLeft, iyTopLeft, ixBottomRight, iyBottomRight, iTransmittance);
                Logging("[Function] BoxOperation (OFF):" + iRet.ToString());
            }
        }

        //---------------------------------------------------------------------
        //* Function Name        : chk_BMPDraw_CheckedChanged        Added Ver5.0.1.0
        //* Overview             : Transparent bitmap image processing
        //---------------------------------------------------------------------
        private void chk_BMPDraw_Click(object sender, EventArgs e)
        {
            int Ret, Id, Command, XPos, YPos, MaskColor, Transmittance;
            String FileName;

            Id = 1;         // ID for management
            XPos = 320;     // X position of displayed transparent
            YPos = 240;     // Y position of displayed transparent
            if (chk_BMPDraw.Checked) {
                // Transparent bitmap
                FileName = textBox1.Text;
                Command = 1;            // Bitmap display
                MaskColor = 0xFF00FF;   // Bitmap's masked color
                Transmittance = 0x7f;   // Transmittance
                Ret = axipropsapiCtrl1.BitmapOperationEx(Id, Command, FileName, XPos, YPos, MaskColor, Transmittance);
                Logging("[Function] BitmapOperationEx (ON," + FileName + "):" + Ret.ToString());
            }
            else {
                // Not transparent bitmap
                Command = 0;            // Bitmap non display
                MaskColor = 0x00;       // Bitmap's masked color
                Transmittance = 0x7f;   // Transmittance
                Ret = axipropsapiCtrl1.BitmapOperationEx(Id, Command, String.Empty, 0, 0, MaskColor, Transmittance);
                Logging("[Function] BitmapOperationEx (OFF):" + Ret.ToString());
            }
        }

        //---------------------------------------------------------------------
        //* Function Name        : DragDraop                          Added Ver5.0.1.0
        //* Overview             : Drop a single file to view
        //---------------------------------------------------------------------
        private void textBox1_DragDrop(object sender, DragEventArgs e)
        {
            String[] Files;
            Files = (String[])e.Data.GetData(DataFormats.FileDrop);
            textBox1.Text = Files[0];
        }

        //---------------------------------------------------------------------
        //* Function Name        : DragEnter                         Added Ver5.0.1.0
        //* Overview             : Drag the files and only valid
        //---------------------------------------------------------------------
        private void textBox1_DragEnter(object sender, DragEventArgs e)
        {
            if (e.Data.GetDataPresent(DataFormats.FileDrop)) {
                e.Effect = DragDropEffects.Copy;
            }
            else {
                e.Effect = DragDropEffects.None;
            }
        }
        
        //---------------------------------------------------------------------
        // Function Name        : OnError
        // Overview             : OnError listener event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnError(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnErrorEvent e)
        {
            //Output Logs
            Logging("[OnError] errorCode[" + e.errorCode.ToString() + "] description[" + e.description + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnDevStatus
        // Overview             : OnDevStatus listener event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnDevStatus(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnDevStatusEvent e)
        {
            //Output Logs
            Logging("[OnDevStatus] channel[" + e.channel + "] status[" + e.status + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnRecStatu
        // Overview             : OnRecStatu listener event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnRecStatus(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnRecStatusEvent e)
        {
            //Output Logs
            Logging("[OnRecStatu] channel[" + e.channel + "] status[" + e.status + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnPlayStatus
        // Overview             : OnPlayStatus listener event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnPlayStatus(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnPlayStatusEvent e)
        {
            //Output Logs
            Logging("[OnPlayStatus] channel[" + e.channel + "] status[" + e.status + "]");
       }

        //---------------------------------------------------------------------
        // Function Name        : OnRecordStatus
        // Overview             : OnRecordStatus listener event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnRecordStatus(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnRecordStatusEvent e)
        {
            //Output Logs
            Logging("[OnRecordStatus] type[" + e.recType + "] timeDate[" + e.timeDate + "] nextRecTime[" + e.nextRecTime + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnImageRefresh
        // Overview             : OnImageRefresh listener event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnImageRefresh(object sender, EventArgs e)
        {
            //Output Logs
            Logging("[OnImageRefresh] No argument.");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnOpStatus
        // Overview             : OnOpStatus listener event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnOpStatus(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnOpStatusEvent e)
        {
            //Output Logs
            Logging("[OnOpStatus] channel[" + e.channel + "] status[" + e.status + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnAlmStatus
        // Overview             : OnAlmStatus listener event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnAlmStatus(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnAlmStatusEvent e)
        {
            //Output Logs
            Logging("[OnAlmStatus] channel[" + e.channel + "] type[" + e.alarmtype + "] timeDate[" + e.timeDate + "] status[" + e.status + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnRecStatusCB
        // Overview             : OnRecStatusCB callback event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnRecStatusCB(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnRecStatusCBEvent e)
        {
            //Output Logs
            Logging("[OnRecStatusCB] status[" + e.status + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnSearchCB
        // Overview             : OnSearchCB callback event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnSearchCB(object sender, EventArgs e)
        {
            //Output Logs
            Logging("[OnSearchCB] Show Search result.");
            ShowResult();
        }

        //---------------------------------------------------------------------
        // Function Name        : OnSearchExCB
        // Overview             : OnSearchExCB callback event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnSearchExCB(object sender, EventArgs e)
        {
            //Output Logs
            Logging("[OnSearchExCB] Show SearchEx result.");
            ShowResultEx();
        }

        //---------------------------------------------------------------------
        // Function Name        : OnPlayStatusCB
        // Overview             : OnPlayStatusCB callback event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnPlayStatusCB(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnPlayStatusCBEvent e)
        {
            //Output Logs
            Logging("[OnPlayStatusCB] status[" + e.status + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnOpStatusCB
        // Overview             : OnOpStatusCB callback event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnOpStatusCB(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnOpStatusCBEvent e)
        {
            //Output Logs
            Logging("[OnOpStatusCB] status[" + e.status + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnAlmStatusCB
        // Overview             : OnAlmStatusCB callback event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnAlmStatusCB(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnAlmStatusCBEvent e)
        {
            //Output Logs
            Logging("[OnAlmStatusCB] status[" + e.status + "]");
        }

        //---------------------------------------------------------------------
        // Function Name        : OnFtpStatusCB
        // Overview             : OnFtpStatusCB callback event
        //---------------------------------------------------------------------
        private void axipropsapiCtrl1_OnFtpStatusCB(object sender, AxIPROPSAPILib._IipropsapiCtrlEvents_OnFtpStatusCBEvent e)
        {
            //Output Logs
            Logging("[OnFtpStatusCB] status[" + e.status + "]");
        }

    }
}